framework/include/auxkernels/MaterialStdVectorRealGradientAux.h

```diff
@@ -0,0 +1,47 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef MATERIALSTDVECTORREALGRADIENTAUX_H
+#define MATERIALSTDVECTORREALGRADIENTAUX_H
+
+// MOOSE includes
+#include "MaterialStdVectorAuxBase.h"
+
+// Forward declarations
+class MaterialStdVectorRealGradientAux;
+
+template<>
+InputParameters validParams<MaterialStdVectorRealGradientAux>();
+
+/**
+ * AuxKernel for outputting a std::vector material-property component to an AuxVariable
+ */
+class MaterialStdVectorRealGradientAux : public MaterialStdVectorAuxBase<RealGradient>
+{
+public:
+
+  MaterialStdVectorRealGradientAux(const std::string & name, InputParameters parameters);
+
+  virtual ~MaterialStdVectorRealGradientAux();
+
+protected:
+
+  ///Compute the value of the material property for the given index and component
```

Comment:
Sorry, maybe I'm really OCD, but we pretty consistently have a space after the ```///``` and only your new comments omit this.